File: scripts/setup-users.py
#!/usr/bin/env python3
"""
Bola Na - User Setup Script
Generates bcrypt password hashes for Sohan and Sandhya
"""

try:
    import bcrypt
except ImportError:
    print("❌ bcrypt not installed")
    print("Install with: pip install bcrypt")
    exit(1)

def generate_hash(password):
    """Generate bcrypt hash for password"""
    return bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt(rounds=10)).decode('utf-8')

# Generate hashes
sohan_hash = generate_hash("Sohan@husband")
sandhya_hash = generate_hash("Sandhya@wifey")

print("🔐 Bola Na - User Setup")
print("=" * 70)
print()

print("📋 Generated Password Hashes:")
print()
print(f"User 1: Sohan")
print(f"  Password: Sohan@husband")
print(f"  Hash: {sohan_hash}")
print()

print(f"User 2: Sandhya")
print(f"  Password: Sandhya@wifey")
print(f"  Hash: {sandhya_hash}")
print()

print("=" * 70)
print()

print("💾 SQL to run in Supabase SQL Editor:")
print()

sql = f"""-- Create users for Bola Na
INSERT INTO users (username, password_hash, display_name, profile_picture)
VALUES 
  ('sohan', '{sohan_hash}', 'Sohan', 'https://api.multiavatar.com/sohan.png'),
  ('sandhya', '{sandhya_hash}', 'Sandhya', 'https://api.multiavatar.com/sandhya.png')
ON CONFLICT (username) DO NOTHING;

-- Verify
SELECT username, display_name FROM users WHERE username IN ('sohan', 'sandhya');
"""

print(sql)
print()

print("✅ Copy the SQL above and run it in Supabase Dashboard")
print("🚀 Then run: npm install && npm run dev")

File: scripts/setup-users.ts
import bcrypt from 'bcryptjs';
import { createClient } from '@supabase/supabase-js';

const SUPABASE_URL = 'https://xntjgqzyytisiuphtibw.supabase.co';
const SUPABASE_ANON_KEY = 'sb_publishable_xKdFZt0fhALc0GBp6pXyMA_qaLJPVSt';

const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY);

async function setupDatabase() {
  try {
    console.log('🚀 Starting database setup...\n');

    // Hash passwords
    const sohanPasswordHash = bcrypt.hashSync('Sohan@husband', 10);
    const sandhyaPasswordHash = bcrypt.hashSync('Sandhya@wifey', 10);

    console.log('✓ Passwords hashed');

    // Insert users
    const { data: users, error: usersError } = await supabase
      .from('users')
      .insert([
        {
          username: 'sohan',
          password_hash: sohanPasswordHash,
          display_name: 'Sohan',
          profile_picture: 'https://api.multiavatar.com/sohan.png',
          is_online: false,
        },
        {
          username: 'sandhya',
          password_hash: sandhyaPasswordHash,
          display_name: 'Sandhya',
          profile_picture: 'https://api.multiavatar.com/sandhya.png',
          is_online: false,
        },
      ])
      .select();

    if (usersError) {
      console.error('❌ Error creating users:', usersError.message);
      return;
    }

    console.log('✓ Users created successfully');
    console.log('\n📋 Created Users:');
    users?.forEach((user) => {
      console.log(`  • ${user.display_name} (@${user.username})`);
    });

    console.log('\n🎉 Database setup complete!');
    console.log('\n📝 Login credentials:');
    console.log('  User 1: sohan / Sohan@husband');
    console.log('  User 2: sandhya / Sandhya@wifey');
  } catch (error) {
    console.error('❌ Setup failed:', error);
  }
}

setupDatabase();
